vncviewer/ServerDialog.cxx

```diff
@@ -153,7 +163,12 @@ void ServerDialog::handleLoad(Fl_Widget *widget, void *data)
   const char* filename = file_chooser->value();
 
   try {
-    dialog->serverName->value(loadViewerParameters(filename));
+    vector<string> serverHistory;
+    dialog->serverName->value(loadViewerParameters(filename,serverHistory));
+    dialog->serverName->clear();
```

Comment:
clear removes all elements from the drop down list. As loadViewerParameters will load the actual server list it will afterwards filled again with the updates list

Reply:
So `clear()` removes the value? In that case we should just ignore the return value and run `loadViewerParameters()` in order to get the updated serverHistory.

In any case, to me it seems like a strange mix to both return data and pass a reference argument for modification? @CendioOssman is this an OK style in your opinion? I'd recommend removing the return value and adding another reference argument for the servername instead.

Reply:
Yes, this could be changed to handle both as return parameters. I just avoided to touch existing code as much as possible to keep the changes small. Will change this together with the comment before in the next commit.

`clear()` remove the elements in the drop down list. The value shown in the label field is set by `value()` and filled by the return from `loadViewerParameters()`